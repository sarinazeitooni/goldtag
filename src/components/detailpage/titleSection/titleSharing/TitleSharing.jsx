import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faSchool} from "@fortawesome/free-solid-svg-icons";
import {BrowserView} from "react-device-detect";
import React from "react";
function TitleSharing({id}){
    const socialMedia = [1,2,3];
    // let socialMedia = new Array(3);
    return(
        <BrowserView>
            <div className='sharing-container'>
                <div>{id.share}</div>
                <div>
                    {
                        socialMedia.map((item)=>{
                            return(
                                <span className="social-media-icon"><FontAwesomeIcon icon={faSchool} /></span>
                            )
                        })
                    }
                </div>
            </div>
        </BrowserView>
    )
};
export default TitleSharing;