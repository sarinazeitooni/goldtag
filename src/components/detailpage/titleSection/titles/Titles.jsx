import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faStar} from "@fortawesome/free-solid-svg-icons";
import titleSectionData from "../Data/titleSectionData";
import {BrowserView} from "react-device-detect";
import React from "react";

function Titles({id , titleSection}) {
    const stars = [1,2,3,4,5];
    return(
        <div>
            <h1 className='title'>{id.title}</h1>
            <h4 className='subtitle'>{id.subtitle}</h4>
            <div className='star-container'>
                <div className='stars'>
                    {stars.map(() => { return (<FontAwesomeIcon icon={faStar} />) })}
                    {titleSectionData.comments}
                </div>
                <BrowserView><div><button className='submit-btn'>{titleSection.shipping}</button></div></BrowserView>
            </div>
        </div>
    )
};
export default Titles;