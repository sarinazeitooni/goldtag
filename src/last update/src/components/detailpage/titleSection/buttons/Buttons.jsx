import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {faClock, faShoppingBag, faShoppingBasket, faStar} from "@fortawesome/free-solid-svg-icons";
import React from "react";
function Buttons({data, titleSectionData}){
    return(
        <div>
            <button className='available-btn'> <FontAwesomeIcon icon={faShoppingBag} />{titleSectionData.available}</button>
            <div className="label-info-container">
                <div className="label-info">
                    <span className='info'><FontAwesomeIcon icon={faClock}/> {data.daynum} </span>
                    {titleSectionData.day}
                </div>
                <div className="label-info">
                    <span className='info'><FontAwesomeIcon icon={faShoppingBasket}/> {data.purchased} </span>
                    {titleSectionData.purchased}
                </div>
            </div>
            <div className='product-charity'>
                با خرید این محصول از گلدتگ <span style={{color:'#ff7a7a'}}>100 تومان</span> به <span style={{color:'#436fbe'}}>خیریه</span> کمک کرده اید.
            </div>
        </div>
    )
};
export default Buttons;