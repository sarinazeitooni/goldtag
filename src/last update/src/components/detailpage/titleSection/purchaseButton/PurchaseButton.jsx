import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faShoppingBag} from "@fortawesome/free-solid-svg-icons";
import NumberFormat from "react-number-format";
import React from "react";
function PurchaseButton({id , data}){
    let discountPercent = ((data.price - data.newprice) * 100) / data.price;
    return(
        <button className='price-container-btn'>
            <div className='shopping-icon'><FontAwesomeIcon icon={faShoppingBag} /></div>
            <div>
                <div className='new-price-container'><span>{id.purchase}</span><span className='price-num'>
                       <NumberFormat className='' value={data.newprice} displayType={'text'}
                                     thousandSeparator={true} renderText={(value, props) =>
                           <span {...props}>{value}</span>
                       }/>

                    </span>
                    <span className='currency'>{id.currency}</span></div>
                <div className='old-price-container'><span className='discount-percent'> {discountPercent}{id.discount}</span>
                    <span className='old-price'><NumberFormat className='' value={data.price} displayType={'text'}
                                                              thousandSeparator={true} renderText={(value, props) =>
                        <span {...props}>{value}</span>
                    }/>{id.currency}</span></div>
            </div>
        </button>
    )
};
export default PurchaseButton;