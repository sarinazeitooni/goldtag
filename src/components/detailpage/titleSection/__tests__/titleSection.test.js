import React from 'react';
import {shallow} from 'enzyme';
import TitleSection from "../../titleSection/titleSection";
import PurchaseButton from "../../titleSection/purchaseButton/PurchaseButton";
import TitleSectionFeatures from "../../titleSection/titleSectionFeatures/titleSectionFeatures";
import TitleSharing from "../../titleSection/titleSharing/TitleSharing";
import Titles from "../../titleSection/titles/Titles";
import Buttons from "../../titleSection/buttons/Buttons";
import SharingContainer from "../../titleSection/sharingContainer/sharing";
describe("SideSection component", () => {
    const container = shallow(<TitleSection data={''}/>)
    it("it should render a div", () => {
        expect(container).toMatchSnapshot();
    });
    it("it should render Titles component", () => {
        expect(container.find(<Titles id={''} titleSection={''} />)).toBeTruthy();
    });
    it("it should render PurchaseButton component", () => {
        expect(container.find(<PurchaseButton data={''} id={''}/>)).toBeTruthy();
    });
    it("it should render Buttons component", () => {
        expect(container.find(<Buttons data={''} titleSectionData={''}/>)).toBeTruthy();
    });
    it("it should render SharingContainer component", () => {
        expect(container.find(<SharingContainer titleSectionData={''}/>)).toBeTruthy();
    });
    it("it should render TitleSectionFeatures component", () => {
        expect(container.find(<TitleSectionFeatures id={''}/>)).toBeTruthy();
    });
    it("it should render TitleSharing component", () => {
        expect(container.find(<TitleSharing id={''}/>)).toBeTruthy();
    });
});