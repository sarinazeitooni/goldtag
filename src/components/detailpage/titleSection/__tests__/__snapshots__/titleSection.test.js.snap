// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideSection component it should render a div 1`] = `ShallowWrapper {}`;
